pom.xml

```diff
@@ -26,10 +26,52 @@
         <maven.compiler.target>17</maven.compiler.target>
         <scala.binary.version>2.13</scala.binary.version>
         <scala.version>${scala.binary.version}.6</scala.version>
-        <spark.version>3.5.3</spark.version>
+        <spark.version>4.1.0-SNAPSHOT</spark.version>
```

Comment:
Any particular reason why we depend on Spark4? Would the latest release version (Spark 3.5.5) work as well?

Reply:
This PR is just meant to take the existing demo code from our spark branch and port it over to the armada-spark repo.  That spark branch was off the 4.1.0-snapshot, so this is too.

I'm working on a separate PR that will address the other versions.  There are non-trivial differences.